pkg/query/iter.go

```diff
@@ -587,3 +593,40 @@ func (it *dedupSeriesIterator) Err() error {
 	}
 	return it.b.Err()
 }
+
+// counterDedupAdjustSeriesIterator is used when we deduplicate counter.
+// It makes sure we always adjust for the latest seen last counter value for all replicas.
```

Comment:
for clarity I think it would be good to elaborate _how_ it adjusts, as in the strategy/heuristic, I think I can tell by reading the code and following the investigation etc., but a comment to confirm would be helpful for me in 3 months :)

Reply:
Yes, good point, will do. Thanks!

Reply:
Done: https://github.com/thanos-io/thanos/pull/2548/commits/e903228a93b93015271ec9a3c5bb3dfd308e4b4e

Does it make sense?